applications/rag/example_notebooks/rag-kaggle-ray-sql-refactored.ipynb

```diff
@@ -0,0 +1,449 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "195d442b-08a9-4469-824c-80188c622c46",
+   "metadata": {},
+   "source": [
+    "# RAG-on-GKE Application\n",
+    "\n",
+    "This is a Python notebook for the RAG on GKE application. For full information, please checkout the GitHub documentation [here](https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/applications/rag/README.md).\n",
+    "\n",
+    "\n",
+    "## Setup Kaggle Credentials\n",
+    "\n",
+    "First we will setup your Kaggle credentials. Replace the following with your own settings from the Kaggle web page. Navigate to https://www.kaggle.com/settings/account and generate an API token to be used to setup the env variable. See https://www.kaggle.com/docs/api#authentication how to create one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "00b1aff4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "KAGGLE_USERNAME = \"<username>\"\n",
+    "KAGGLE_KEY = \"<token>\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "44dca61a-4544-4bbc-b78c-9e0805b96192",
+   "metadata": {},
+   "source": [
+    "## Installing Prerequisites\n",
+    "\n",
+    "To run this notebook, we will install the following tools and binaries:\n",
+    "* Ray\n",
+    "* Kaggle\n",
+    "* Langchain\n",
+    "* CloudSQL"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a814e91b-3afe-4c28-a3d6-fe087c7af552",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install ray[default]==2.9.3 kaggle==1.6.6\n",
+    "!pip install langchain==0.1.10 ray==2.9.3 datasets sentence-transformers\n",
```

Comment:
I think there were some issues in the past with some of these packages taking way too long to install. Specifically sentence-transformers. Are we going to bake the dependencies into the jupyter image?

Reply:
I've tested and it turns out the notebook does need to install langchain and sentence-transformers. I can see if we could use a custom image here to skip these pip installs.

Reply:
Changed to a custom image with dependencies baked in. Removed this section from the notebook.